src/digitalIdentity-frontend/src/app/pages/login-page/login-page.component.spec.ts

```diff
@@ -57,4 +72,42 @@ describe('LoginPageComponent', () => {
     let test_div = de.query(By.css('.test-card'));
     expect(test_div).toBeNull();
   });
+
+  it('should have an initial state like', () => {
+    expect(de.query(By.css('input'))).toBeDefined();
+    expect(component.formGroup.invalid).toBeTruthy();
+  });
+
+  it('empty input form is invalid', (() => {
+    let form = component.formGroup;
+    expect(fixture.debugElement.nativeElement.querySelector('input').innerHTML).toBe("");
+    expect(form.invalid).toBeTruthy();
+  }));
+
+  it('should create a form with email, password field and a button', () => {
+    let email = fixture.debugElement.nativeElement.querySelector('input');
+    let password = fixture.debugElement.nativeElement.querySelector('input');
```

Comment:
password and email query the same Element

Reply:
Mistake. I forgot to specify the right element. Thanks for your points! I will work on it again